src/runtime/virtcontainers/network_darwin.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2023 Apple Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+
+	persistapi "github.com/kata-containers/kata-containers/src/runtime/virtcontainers/persist/api"
+)
+
+var endpointNotSupported error = errors.New("Unsupported endpoint on Darwin")
+
+// DarwinNetwork represents a sandbox networking setup.
+type DarwinNetwork struct {
+	networkID         string
+	interworkingModel NetInterworkingModel
+	networkCreated    bool
+	eps               []Endpoint
+}
+
+func NewNetwork(configs ...*NetworkConfig) (Network, error) {
+	if len(configs) > 1 {
+		return nil, fmt.Errorf("Too many network configurations")
+	}
+
+	// Empty constructor
+	if len(configs) == 0 {
+		return &DarwinNetwork{}, nil
+	}
+
+	config := configs[0]
+	if config == nil {
+		return nil, fmt.Errorf("Missing network configuration")
```

Comment:
Prefer `errors.New` if no formatting (also go idioms is to start errors lowercased)